lib/class-wp-theme-json-gutenberg.php

```diff
@@ -3578,4 +3578,49 @@ protected function get_feature_declarations_for_node( $metadata, &$node ) {
 
 		return $declarations;
 	}
+
+	/**
+	 * Generates style declarations for a node's features e.g. color, border,
+	 *
+	 * @since 6.3.0
+	 * @param array $styles
+	 * @param array $vars
+	 * @return array
+	 */
+	private static function convert_variables_to_value( $styles, $vars ) {
+		foreach ( $styles as $key => $style ) {
+			if ( is_array( $style ) ) {
+				$styles[ $key ] = self::convert_variables_to_value( $style, $vars );
+			}
+			if ( is_string( $style ) && 0 === strpos( $style, 'var(' ) ) {
+				$styles[ $key ] = isset( $vars[ $style ] ) ? $vars[ $style ] : $style;
+			}
+		}
+
+		return $styles;
+	}
+
+	/**
+	 * Get styles defined in theme.json with values for the css variables.
+	 *
+	 * @since 6.3.0
+	 * @return array
+	 */
+	public function get_styles_with_values() {
+		$preset_vars = static::compute_preset_vars( self::get_settings(), static::VALID_ORIGINS );
+		$theme_vars  = static::compute_theme_vars( self::get_settings() );
+		$vars        = array_reduce(
+			array_merge( $preset_vars, $theme_vars ),
```

Comment:
@oandregal I'd like to know your idea on this, would this produce any unseen problem?

Reply:
Presets and custom properties defined by the theme and the only source of data we have available, so this is fine.